management/guardian_test.go

```diff
@@ -13,266 +14,312 @@ func TestGuardian(t *testing.T) {
 	t.Run("MultiFactor", func(t *testing.T) {
 		t.Run("List", func(t *testing.T) {
 			mfa, err := m.Guardian.MultiFactor.List()
-			if err != nil {
-				t.Error(err)
-			}
-			t.Logf("%v\n", mfa)
+			assert.NoError(t, err)
+			assert.Greater(t, len(mfa), 1)
 		})
 
 		t.Run("Policy", func(t *testing.T) {
-			// Has to be one of "all-applications" or "confidence-score", but not both.
-			// If omitted, it removes all policies.
-			err := m.Guardian.MultiFactor.UpdatePolicy(&MultiFactorPolicies{"all-applications"})
-			if err != nil {
-				t.Error(err)
-			}
-			p, err := m.Guardian.MultiFactor.Policy()
-			if err != nil {
-				t.Error(err)
-			}
-			t.Logf("%v\n", p)
+			initialPolicy, err := m.Guardian.MultiFactor.Policy()
+			assert.NoError(t, err)
+
+			// Has to be one of "all-applications" or "confidence-score",
+			// but not both. If omitted, it removes all policies.
+			expectedPolicy := &MultiFactorPolicies{"all-applications"}
+			err = m.Guardian.MultiFactor.UpdatePolicy(expectedPolicy)
+			assert.NoError(t, err)
+
+			actualPolicy, err := m.Guardian.MultiFactor.Policy()
+			assert.NoError(t, err)
+			assert.Equal(t, expectedPolicy, actualPolicy)
+
+			err = m.Guardian.MultiFactor.UpdatePolicy(initialPolicy)
+			assert.NoError(t, err)
 		})
 
 		t.Run("Phone", func(t *testing.T) {
 			t.Run("Provider", func(t *testing.T) {
-				err := m.Guardian.MultiFactor.Phone.UpdateProvider(&MultiFactorProvider{Provider: auth0.String("phone-message-hook")})
-				if err != nil {
-					t.Error(err)
-				}
-				p, _ := m.Guardian.MultiFactor.Phone.Provider()
-				t.Logf("%v\n", p)
+				initialProvider, err := m.Guardian.MultiFactor.Phone.Provider()
+				assert.NoError(t, err)
+
+				expectedProvider := &MultiFactorProvider{Provider: auth0.String("phone-message-hook")}
+
+				err = m.Guardian.MultiFactor.Phone.UpdateProvider(expectedProvider)
+				assert.NoError(t, err)
+
+				actualProvider, err := m.Guardian.MultiFactor.Phone.Provider()
+				assert.NoError(t, err)
+				assert.Equal(t, expectedProvider, actualProvider)
+
+				err = m.Guardian.MultiFactor.Phone.UpdateProvider(initialProvider)
+				assert.NoError(t, err)
 			})
 
 			t.Run("Enable", func(t *testing.T) {
-				err := m.Guardian.MultiFactor.Phone.Enable(false)
-				if err != nil {
-					t.Error(err)
-				}
+				initialStatus, err := getInitialMFAStatus("sms")
+				assert.NoError(t, err)
+
+				t.Cleanup(func() {
```

Comment:
Just curious, why use a cleanup method here and not in the other cases?

Reply:
Oh, I see some other test cases use the cleanup method, but there are ones that aren't. Is this intended?

Reply:
Good catch, they should all consistently use the cleanup method at the beginning of the func. Fixed in [69f7d42](https://github.com/auth0/go-auth0/pull/53/commits/69f7d4270c8b1942f8ec21cc56f50ec518bdb5e6)

Also should have probably attached the cleanup on all the givens on previous tests as well to simplify the test logic. Will do a final sweep once all resources tests are updated to clean this up.

Reply:
e.g. https://github.com/auth0/go-auth0/pull/53/files#diff-eeb45118a10f1f40839fd8728d64f3ba285a252bc385d9e73c7159529fe68110R481-R483

Reply:
Yeah, good idea.